From fba0e9aa85f670373b2d65cf170dacb00228d0b7 Mon Sep 17 00:00:00 2001
From: leo <leo233.huang@gmail.com>
Date: Sat, 12 Apr 2025 22:30:48 +0800
Subject: [PATCH] rust for linux patch


diff --git a/linux/linux.mk b/linux/linux.mk
index f915823d..790054bb 100644
--- a/linux/linux.mk
+++ b/linux/linux.mk
@@ -161,11 +161,12 @@ endif
 # Disable building host tools with -Werror: newer gcc versions can be
 # extra picky about some code (https://bugs.busybox.net/show_bug.cgi?id=14826)
 LINUX_MAKE_FLAGS = \
-	HOSTCC="$(HOSTCC) $(subst -I/,-isystem /,$(subst -I /,-isystem /,$(HOST_CFLAGS))) $(HOST_LDFLAGS)" \
+	HOSTCC=/usr/bin/gcc  \
 	ARCH=$(KERNEL_ARCH) \
 	KCFLAGS="$(LINUX_CFLAGS)" \
 	INSTALL_MOD_PATH=$(TARGET_DIR) \
-	CROSS_COMPILE="$(TARGET_CROSS)" \
+	LLVM=1 \
+	CC="/usr/bin/clang" \
 	WERROR=0 \
 	REGENERATE_PARSERS=1 \
 	DEPMOD=$(HOST_DIR)/sbin/depmod
@@ -184,9 +185,9 @@ endif
 # abusing those aliases for system call entry points, in order to
 # sanitize the arguments passed from user space in registers.
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=82435
-ifeq ($(BR2_TOOLCHAIN_GCC_AT_LEAST_8),y)
-LINUX_CFLAGS += -Wno-attribute-alias
-endif
+# ifeq ($(BR2_TOOLCHAIN_GCC_AT_LEAST_8),y)
+# LINUX_CFLAGS += -Wno-attribute-alias
+# endif
 
 # Disable FDPIC if enabled by default in toolchain
 ifeq ($(BR2_BINFMT_FDPIC),y)
-- 
2.34.1

